daft/catalog/__init__.py

```diff
@@ -371,23 +389,58 @@ def _from_obj(obj: object) -> Table:
         """Returns a Daft Table from a supported object type or raises an error."""
         raise ValueError(f"Unsupported table type: {type(obj)}")
 
-    # TODO catalog APIs part 3
-    # @property
-    # @abstractmethod
-    # def name(self) -> str:
-    #     """Returns the table name."""
-
-    # TODO catalog APIs part 3
-    # @property
-    # @abstractmethod
-    # def inner(self) -> object | None:
-    #     """Returns the inner table object if this is an adapter."""
+    ###
+    # read methods
+    ###
 
     @abstractmethod
-    def read(self) -> DataFrame:
+    def read(self, **options) -> DataFrame:
         """Returns a DataFrame from this table."""
 
+    def select(self, *columns: ColumnInputType) -> DataFrame:
+        """Returns a DataFrame from this table with the selected columns."""
+        return self.read().select(*columns)
+
+    def show(self, n: int = 8) -> None:
+        """Shows the first n rows from this table."""
+        return self.read().show(n)
+
+    ###
+    # write methods
+    ###
+
+    @abstractmethod
+    def write(self, df: DataFrame, mode: str = "append", **options) -> None:
+        """Writes the DataFrame to this table.
+
+        Args:
+            df (DataFrame): datafram to write
+            mode (str): write mode such as 'append' or 'overwrite'
+            **options: additional format-dependent write options
+        """
+
+    def append(self, df: DataFrame, **options) -> None:
+        """Appends the DataFrame to this table.
+
+        Args:
+            df (DataFrame): dataframe to append
+            **options: additional format-dependent write options
+        """
+        self.write(df, mode="append", **options)
+
+    def overwrite(self, df: DataFrame, **options) -> None:
```

Comment:
Hmm, what's the reasoning behind having these as dedicated methods? Most libraries have them as part of the `write` method

```py
write(df, mode="append")
write(df, mode="overwrite")
```

Reply:
The libraries we work with expose these in addition to the `.write` with mode flag.

* PyIceberg - [append](https://py.iceberg.apache.org/reference/pyiceberg/table/#pyiceberg.table.Table.append), [ovewrite](https://py.iceberg.apache.org/reference/pyiceberg/table/#pyiceberg.table.Table.overwrite)
* Spark - [append](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.DataFrameWriterV2.append.html#pyspark.sql.DataFrameWriterV2.append), [overwrite](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.DataFrameWriterV2.overwrite.html#pyspark.sql.DataFrameWriterV2.overwrite)